### PR TITLE
make compatibel for other detectors, e.g. ILD

### PR DESCRIPTION
**Pull request by gaede**
BEGINRELEASENOTES
- make compatible for other detectors, e.g. ILD
     - introduce new parameters with indices of tracker hit collections:
          - AllHitCollectionIndices
          - TrackerHitCollectionIndices
      - default values as used for CLIC
      - protect against empty collections

ENDRELEASENOTES

### CODE REVIEW

**Comment by andresailer** [comment]
You need to rebase and also fix the formatting. And I still need to see if the changes have any side effects. Probably not before Easter.

**Comment by gaede** [comment]
I did the rebase and there are no compiler errors. Will merge this now as I need a release tag with these changes in. 
In the unlikely case you find a problem with this you can always roll-back ...
Of course we tested it before creating the PR.

**Comment by petricm** [comment]
Why did you merge if the CI is failing? Please fix it.

**Comment by gaede** [comment]
Marko, the CI with the builds and tests is fine - what is failing is the formatting. I have no idea how to fix it as there are no error messages nor is there any documentation on what the format should be.
If I look at the diff's I have kept the original format (most of it is copy and paste). 
I don't think this  format CI is helping code quality or productivity in any way - why don't you take it out again ?
What I have fixed is 'sloppy code' with hard coded array initializers that I have turned into processor parameters as it it should have been from the start. This is needed to run the processor for ILD or any other detector that has a different tracker layout than CLICdp. All our tracking code is made 'detector agnostic' and so should this one be.

**Comment by petricm** [comment]
The formatting can be applied by invoking the build target `format` and it can be checked by `check-format`, I don't think this is too hard...

**Comment by gaede** [comment]
```
HEAD/ConformalTracking/HEAD/build (master)$ make format
make: *** No rule to make target `format'.  Stop.
```

**Comment by petricm** [comment]
https://github.com/iLCSoft/ConformalTracking/blob/master/cmake/clang-cpp-checks.cmake#L10-L36

If you are on a Mac, `clang-format` is not installed as part of Apple LLVM...

File: src/ConformalTracking.cc [hunk 1]
```diff
@@ -913,7 +920,10 @@ void ConformalTracking::combineCollections(SharedKDClusters& kdClusters, UKDTree
 
   // Loop over all given collections
   for (unsigned int i = 0; i < combination.size(); i++) {
+
     // Copy the clusters to the output vector
+    if( collectionClusters.size() <= unsigned( combination[i] )  ) continue; // protect against empty subdetectors
```
Comment (andresailer) [inline src/ConformalTracking.cc]: This check is wrong, the size of the map doesn't say anything about which key is present.

**Pull request merged**